innovation.game.php

```diff
@@ -8234,6 +8235,37 @@ function stPlayerInvolvedTurn() {
                 $step_max = 1;
                 break;
 
+            // id 179, Artifacts age 7: International Prototype Metre Bar
+            case "179N1":
+                $step_max = 1;
+                break;
+
+            // id 180, Artifacts age 7: Hansen Writing Ball
+            case "180C1":
+                // I compel you to draw four 7's!
+                self::executeDraw($player_id, 7);
+                self::executeDraw($player_id, 7);
+                self::executeDraw($player_id, 7);
+                self::executeDraw($player_id, 7);
+                
+                $step_max = 2;
+                break;
+
+            case "180N1":
+                $stop = false;
+                do {
+                    $card = self::executeDraw($player_id, 7, 'revealed'); // "Draw and reveal a 7. 
+                    if ($card['spot_1'] == 6 || $card['spot_2'] == 6 || $card['spot_3'] == 6 || $card['spot_4'] == 6) {
```

Comment:
I updated this part to use this existing function: `self::hasRessource($card, 6)`